### ISSUE TITLE
Exported variables not immediately accessible in the inspector upon modifying .gd script

### ISSUE THREAD

**Issue by kKapalka** [godotengine/godot#85135]
### Godot version

4.1.1

### System information

Godot v4.1.1.stable - Windows 10.0.19045 - Vulkan (Forward+) - dedicated Radeon RX 5500 XT (Advanced Micro Devices, Inc.; 31.0.12027.9001) - AMD Ryzen 5 3500X 6-Core Processor (6 Threads)

### Issue description

Exported variables seem to not be immediately accessible in Godot Inspector. Meaning that if a new variable is added, I am forced to restart my project before I can access it.

The issue seems kinda important, because if multiple people work on the same project, then when person A introduces new exported variables and sets them to some value, then person B has to: close the project, pull from remote, and start the project once again, otherwise they risk setting all those exported variables to null.

### Steps to reproduce

Reproduction steps:
1 - Have a node with .gd script attached present on the scene
2 - Add new `@export` var to the script, and save it.

(issue has been noticed while using PackedScenes with .gd scripts attached, issue may be present in other nodes as well)

Expected behavior: I can immediately edit the exported variable for the node that is present on the scene.
Actual behavior: The new exported variable remains invisible. I am forced to restart the project entirely before I can access this variable.

### Minimal reproduction project

N/A

**Comment by mieldepoche** [comment]
iirc 4.1 used to sometimes inspect the edited script in the inspector rather than the selected scene's node, so it lead to me thinking the variables didn't show up consistently. Like, adding a new script to a node sneakily stole the inspector.
After testing in 4.2, none of the problems mentionned are here. Also you don't need to save for exported variables to appear in the inspector.

**Comment by AThousandShips** [comment]
Can you please try this with a supported version, like 4.1.3 (only the latest patch version is supported, and it might already be fixed)

**Comment by kKapalka** [comment]
![image](https://github.com/godotengine/godot/assets/23338566/3f2a70e4-972a-48c2-b116-50164d83bd7d)
@AThousandShips - the issue persists in Godot 4.1.3

Added new variable to a script attached to PackedScene node on the scene and saved the script. Then clicked over all the nodes of that PackedScene type.

New export variable is nowhere to be found. Only after restarting the project it appears again.

**Comment by AThousandShips** [comment]
Okay your description didn't make it clear it was an inherited class, then this is a duplicate of:
* https://github.com/godotengine/godot/issues/82998
